### PR TITLE
Google Analytics: TrackAdmin, AnonymizeIp, and TrackUserID Support

### LINKED ISSUE #2782: Google Analytics Connector - Track Admins Always True
## Description of bug
When setting the 'Track for Administrators' checkbox to false and pressing Save within the Google Analytics Connector the form is automatically switching the checkbox to true when saving.

Additionally, if you modify the underlying GoogleAnalytics.config file in the portal root folder to have this setting set to false the UI is still presenting it as checked to true.

## Steps to reproduce

1. Go to Settings > Connectors > Google Analytics

2. Ensure a tracking code is added

3. Change Track For Admins to false

3. Press save

The settings are saved and the Track for Admins appears to be set to true.

## Expected result

The Track for Admins option should save properly depending upon the users selection.

## Affected version
* [x] 9.3.2

## Affected browser

* [x ] Chrome
* [x ] Edge

Initial tests were completed in the above browsers.

### PR DESCRIPTION
**Pull request by ohine**
## Summary
This PR fixes #2782 and also implements `AnonymizeIp` and `TrackUserID` settings in Google Analytics.

### CODE REVIEW

File: DNN Platform/Connectors/GoogleAnalytics/GoogleAnalyticsConnector.cs [hunk 1]
```diff
@@ -158,7 +160,8 @@ public bool SaveConfig(int portalId, IDictionary<string, string> values, ref boo
 
                     trackingID = values["TrackingID"] != null ? values["TrackingID"].ToString().Trim() : String.Empty;
                     urlParameter = values["UrlParameter"] != null ? values["UrlParameter"].ToString().Trim() : String.Empty;
-                    trackForAdmin = bool.TryParse(values["TrackAdministrators"].ToLowerInvariant(), out trackForAdmin);
+
+                    bool.TryParse(values["TrackAdministrators"].ToLowerInvariant(), out trackForAdmin);
```
Comment (bdukes) [inline DNN Platform/Connectors/GoogleAnalytics/GoogleAnalyticsConnector.cs]: 👍

File: DNN Platform/Connectors/GoogleAnalytics/Scripts/connector.js [hunk 2]
```diff
@@ -105,7 +106,7 @@ define(["jquery", "knockout", "templatePath/scripts/config", "templatePath/scrip
 
                         // Set the isDeactivating flag to true to override the default save behaviour
                         // Temporary workaround until delete functionality on connectors is improved
-                        conn.configurations[3].value("true");
+                        conn.configurations[5].value("true");
```
Comment (bdukes) [inline DNN Platform/Connectors/GoogleAnalytics/Scripts/connector.js]: Is there any way to get away from mysterious numbers defining these values?

File: DNN Platform/Connectors/GoogleAnalytics/connector.htm [hunk 3]
```diff
@@ -15,6 +15,16 @@
...
     </div>
 
+    <div>
+        <label for="chkAnonymizeIp" data-bind="html: configurations[3].localizedName"></label>
+        <input id="chkAnonymizeIp" type="checkbox" data-bind="checked: configurations[3].value, checkedValue: configurations[3].value" />
+    </div>
+
+    <div>
+        <label for="chkTrackUserId" data-bind="html: configurations[4].localizedName"></label>
+        <input id="chkTrackUserId" type="checkbox" data-bind="checked: configurations[4]].value, checkedValue: configurations[4].value" />
```
Comment (bdukes) [inline DNN Platform/Connectors/GoogleAnalytics/connector.htm]: Looks like a syntax error: `checked: configurations[4]]`

Comment (bdukes) [reply DNN Platform/Connectors/GoogleAnalytics/connector.htm]:
```suggestion
        <input id="chkTrackUserId" type="checkbox" data-bind="checked: configurations[4].value, checkedValue: configurations[4].value" />
```

File: DNN Platform/Library/Services/Analytics/GoogleAnalyticsEngine.cs [hunk 4]
```diff
@@ -94,5 +94,52 @@ public override string RenderScript(string scriptTemplate)
...
+
+                var sb = new System.Text.StringBuilder();
+                if (anonymize)
+                {
+                    sb.Append("ga('set', 'anonymizeIp', true);");
+                }
+
+                if (trackingUserId)
+                {
+                    sb.AppendFormat("ga('set', 'userId', {0});", UserController.Instance.GetCurrentUserInfo().UserID);
```
Comment (bdukes) [inline DNN Platform/Library/Services/Analytics/GoogleAnalyticsEngine.cs]: should this check if they're logged in?  or do we want to track `-1`?

File: DNN Platform/Connectors/GoogleAnalytics/GoogleAnalyticsConnector.cs [hunk 5]
```diff
@@ -117,6 +117,8 @@ public bool HasConfig(int portalId)
                 { "TrackingID", trackingId },
                 { "UrlParameter", urlParameter},
                 { "TrackAdministrators", trackForAdmin.ToString()},
+                { "AnonymizeIp", trackForAdmin.ToString()},
+                { "TrackUserId", trackForAdmin.ToString()},
```
[the remark this answers is not in the document]
Comment (ohine) [reply DNN Platform/Connectors/GoogleAnalytics/GoogleAnalyticsConnector.cs]: That’s what I get for rushing. I’ll get these fixed shortly. Thanks for the review!

File: DNN Platform/Library/Services/Analytics/GoogleAnalyticsEngine.cs (hunk 4 above)
Comment (ohine) [reply DNN Platform/Library/Services/Analytics/GoogleAnalyticsEngine.cs]: I grabbed that code from my old GA extension, and it was running for all users in that extension. 

However, I can see pros/cons for either way. I'm good with -1 for guests or adding a check  IsAuthenticated prior to setting the UserId var in GA. Whichever sounds best to the @dnnsoftware/approvers i'll make happen with another commit.

File: DNN Platform/Connectors/GoogleAnalytics/Scripts/connector.js (hunk 2 above)
Comment (ohine) [reply DNN Platform/Connectors/GoogleAnalytics/Scripts/connector.js]: Agreed, It's something I'd also like to fix, but I didn't have time this morning to look into the configuration property in the connector engine. I'm looking at it now to see if we can remove those and replace them with the name of the setting.

**Pull request merged**